llvm/lib/Analysis/MemoryProfileInfo.cpp

```diff
@@ -222,6 +228,15 @@ void CallStackTrie::collectContextSizeInfo(
     collectContextSizeInfo(Caller.second, ContextSizeInfo);
 }
 
+void CallStackTrie::convertHotToNotCold(CallStackTrieNode *Node) {
+  if (Node->AllocTypes & static_cast<uint8_t>(AllocationType::Hot)) {
+    Node->AllocTypes &= ~static_cast<uint8_t>(AllocationType::Hot);
+    Node->AllocTypes |= static_cast<uint8_t>(AllocationType::NotCold);
```

Comment:
Can we add some helpers to Node so that we don't have to sprinkle static_cast and bit masking operations in different places?

Reply:
will do

Reply:
done